src/tn/eduVision/services/SallesService.java

```diff
@@ -0,0 +1,194 @@
+ */
+package tn.eduVision.services;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import tn.eduVision.entités.Salle;
+import tn.eduVision.entités.TypeSalle;
+import tn.eduVision.exceptions.NoDataFoundException;
+import tn.eduVision.tools.CustomLogger;
+import tn.eduVision.tools.SqlConnectionManager;
+
+/**
+ *
+ * @author job_j
+ */
+public class SallesService implements Iservices<Salle> {
+
+    private final Connection _connection = SqlConnectionManager.getConnection();
+    private static final Logger _logger = CustomLogger.getInstance().getLogger();
+    private static PreparedStatement statement = null;
+    @Override
+    public void add(Salle salle) {
+        PreparedStatement statement = null;
+        try{
+            
+            String insertSalle = "INSERT INTO `ressources` "
+                    + "(`type_ressource`, `type_salle`, `nom_salle`, `capacite`, `equipements`, `disponibilite`) "
+                    + "VALUES (?, ?, ?, ?, ?, ?);";
+            statement = _connection.prepareStatement(insertSalle);
+            statement.setString(1, salle.getTypeRessource().name());
+            statement.setString(2, salle.getTypeSalle().toString());
+            statement.setString(3, salle.getNomSalle());
+            statement.setInt(4, salle.getCapacite());
+            statement.setString(5, salle.getEquipements());
+            statement.setString(6, salle.getDisponibilite());
+            
+            statement.executeUpdate();
+            
+            
+        }
+        catch(SQLException ex){
+            _logger.log(Level.SEVERE, ex.getMessage());
+            
+        }
+        
+    }
+
+    @Override
+    public void update(Salle salle) {
+        try{
+            
+            String updateSalle = "UPDATE `ressources` SET `nom_salle` = ? , `equipements` = ?, `type_salle` = ? , `capacite` = ?, `type_ressource` = ?, `disponibilite` = ?"
+                    + "WHERE `ressources`.`id_ressource` = ?;";
+            PreparedStatement statement = _connection.prepareStatement(updateSalle);
+            statement.setString(1, salle.getNomSalle());
+            statement.setString(2, salle.getEquipements());
+            statement.setString(3, salle.getTypeSalle().name());
+            statement.setInt(4, salle.getCapacite());
+            statement.setString(5, salle.getTypeRessource().name());
+            statement.setString(6, salle.getDisponibilite());
+            statement.setInt(7, salle.getIdRessource());
+            statement.executeUpdate();
+        }
+        catch(SQLException ex){
+            _logger.log(Level.SEVERE, ex.getMessage());
+        }
+        finally{
+            CloseStatment(statement);
+        }
+    }
+
+    @Override
+    public void delete(Salle salle) {
```

Comment:
this is actually what i wanted to discusss , some entities have Lists that are fetched from the database , if we go with this approach (elli chofneh deja fil classe) it needs a workaround (like initilaizing the lists as empty lists) or we can make a dto for creating and updating and for deleting we just use the id. What do you think ?

Reply:
make a PR warini example also if you can

Reply:
also i think we should always keep method te3hom and keep it simple ama as u wish